createtopics.go

```diff
@@ -0,0 +1,218 @@
+package kafka
+
+import (
+	"bufio"
+	"time"
+)
+
+type ConfigEntry = createTopicsRequestV2ConfigEntry
+
+type createTopicsRequestV2ConfigEntry struct {
+	ConfigName  string
+	ConfigValue string
+}
+
+func (t createTopicsRequestV2ConfigEntry) size() int32 {
+	return sizeofString(t.ConfigName) +
+		sizeofString(t.ConfigValue)
+}
+
+func (t createTopicsRequestV2ConfigEntry) writeTo(w *bufio.Writer) {
+	writeString(w, t.ConfigName)
+	writeString(w, t.ConfigValue)
+}
+
+type ReplicaAssignment = createTopicsRequestV2ReplicaAssignment
+
+type createTopicsRequestV2ReplicaAssignment struct {
+	Partition int32
+	Replicas  int32
+}
+
+func (t createTopicsRequestV2ReplicaAssignment) size() int32 {
+	return sizeofInt32(t.Partition) +
+		sizeofInt32(t.Replicas)
+}
+
+func (t createTopicsRequestV2ReplicaAssignment) writeTo(w *bufio.Writer) {
+	writeInt32(w, t.Partition)
+	writeInt32(w, t.Replicas)
+}
+
+type TopicConfig = createTopicsRequestV2Topic
```

Comment:
Can we make this its own type instead of aliasing to an internal type? Go type aliases were designed to facilitate code migrations, this could mislead the reader into interpreting that the code is being refactored.

Unexported types also aren't exposed by godoc, which means the reader needs to go look at the source to understand how it can use the code.

Reply:
Good points. I'll see if I can clean this up a bit.

My intent was to preserve the `*RequestV2Topic` types because there's API compatibility information encoded in the type names. However, I can probably accomplish the same thing with an alias in the opposite direction after making the exposed types first class.